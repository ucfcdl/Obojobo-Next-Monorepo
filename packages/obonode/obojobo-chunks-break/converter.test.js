import Converter from './converter'

describe('Break Converter', () => {
	test('slateToObo converts a Slate node to an OboNode with content', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => ({})
			}
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: { width: 'large', triggers: 'mock-triggers' }
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node and sets a default width', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {} // not sending width here
		}
		const slateNode = Converter.oboToSlate(oboNode)
		expect(slateNode).toHaveProperty('data.content.width', 'normal')
	})
})
