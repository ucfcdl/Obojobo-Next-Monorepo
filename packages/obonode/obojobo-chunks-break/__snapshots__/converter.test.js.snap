// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Break Converter oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "data": Object {
    "content": Object {
      "triggers": "mock-triggers",
      "width": "large",
    },
  },
  "key": "mockKey",
  "object": "block",
  "type": "mockType",
}
`;

exports[`Break Converter slateToObo converts a Slate node to an OboNode with content 1`] = `
Object {
  "children": Array [],
  "content": Object {},
  "id": "mockKey",
  "type": "mockType",
}
`;
